loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,647 @@
+        if(name == NULL)
+        {
+            lmLogError(gSQLiteGroup, "Invalid index for getParameterName in database: %s", parentDB->getDBName());
+        }
+        return name;
+    }
+
+    int getParameterIndex(const char* name)
+    {
+        int index = sqlite3_bind_parameter_index(statementHandle, name);
+        if(name == NULL)
+        {
+            lmLogError(gSQLiteGroup, "Invalid name for getParameterIndex in database: %s", parentDB->getDBName());
+        }
+        return index;
+    }
+
+    int bindInt(int index, int value)
+    {
+        int result = sqlite3_bind_int(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindInt for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindDouble(int index, double value)
+    {
+        int result = sqlite3_bind_double(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindDouble for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindString(int index, const char *value)
+    {
+        int result = sqlite3_bind_text(statementHandle, index, value, -1, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindBytes(int index, utByteArray *value)
+    {
+        void *bytes;
+        int size;
+        int result;
+
+        if(!value || !value->getSize())
+        {
+            bytes = NULL;
+            size = 0;
+        }
+        else
+        {
+            bytes = value->getDataPtr();      
+            size = (int)value->getSize();
+        }
+
+        //bind the blob to the statement
+        result = sqlite3_bind_blob(statementHandle, index, (const void *)bytes, size, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindBytes for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;
+    }
+
+    int step()
+    {
+        int result = sqlite3_step(statementHandle); 
+        if(result == SQLITE_ERROR)
+        {
+            lmLogError(gSQLiteGroup, "Error calling step for database: %s", parentDB->getDBName());
+        }
+        return result;
+    }
+
+    bool stepAsync()
+    {
+        if(asyncStepInProgress)
+        {
+            lmLogError(gSQLiteGroup, "Attempting to run multiple stepAsync calls on the same statement for database: %s", parentDB->getDBName());
+            return false;
+        }
+
+        //set up the progress handler
+        sqlite3_progress_handler(parentDB->dbHandle, 
```

Comment:
It should be as the thread and logic is all still per statement. Might chug
if too many, but that's the user's responsibility just like with any
threading, no?

Alternatives?
On 20 Feb 2015 2:19 AM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r25040058>:
>
> > +        {
> > +            lmLogError(gSQLiteGroup, "Error calling step for database: %s", parentDB->getDBName());
> > +        }
> > +        return result;
> > +    }
> > +
> > +    bool stepAsync()
> > +    {
> > +        if(asyncStepInProgress)
> > +        {
> > +            lmLogError(gSQLiteGroup, "Attempting to run multiple stepAsync calls on the same statement for database: %s", parentDB->getDBName());
> > +            return false;
> > +        }
> > +
> > +        //set up the progress handler
> > +        sqlite3_progress_handler(parentDB->dbHandle,
>
> Since this is per-db it will be called many times. Is that safe?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r25040058>.
>